### PR TITLE
Recsys ranking fit

### PR DESCRIPTION
**Pull request by haru-256**
This pull request introduces a configuration-driven training pipeline for the recsys-ranking project, adds modular model configuration files, and improves the flexibility and clarity of the DeepFM and DIN model implementations. The changes enable easier experimentation with different model architectures and training settings using Hydra, and provide better documentation and maintainability for the codebase.

**Configuration and Experiment Management:**

* Added Hydra-based configuration system, including `config.yaml`, model-specific configs (`deepfm.yaml`, `dlrm.yaml`, `din.yaml`), and Hydra defaults, enabling flexible experiment management and reproducibility. (`projects/recsys-ranking/src/config/config.yaml` [[1]](diffhunk://#diff-7aab2977f81529ba8f17606d3e8b5ebf7acc2f696cdaecb4089d89c027a7926fR1-R35) `projects/recsys-ranking/src/config/model/deepfm.yaml` [[2]](diffhunk://#diff-947e97e3a43a3d0962c1540230b487643a971202adcc979c8ea44d7115c08674R1-R7) `projects/recsys-ranking/src/config/model/dlrm.yaml` [[3]](diffhunk://#diff-5fed7ad4c35a08a0dd9711f50e45d2e34e91e23cb16826f1a07cd23fa8d1cd9eR1-R22) `projects/recsys-ranking/src/config/model/din.yaml` [[4]](diffhunk://#diff-e957862991bc6ba4710081139029a47f01f590c73b0e1b67c8e97a30b4c33118R1-R20) `projects/recsys-ranking/src/config/hydra/default.yaml` [[5]](diffhunk://#diff-d0f5544272a1d5bb0d5d58f9ed705ab133bce7f505ebc63c238139c17338532cR1-R7)

* Added `.gitignore` rule to exclude `results` directory, which is now used for experiment outputs. (`projects/recsys-ranking/.gitignore` [projects/recsys-ranking/.gitignoreR1](diffhunk://#diff-7cb20c20c5c5280c58c838cbabf5e48164577864ce17cf428a7e08f6489ee5dfR1))

**Training Pipeline:**

* Implemented a new `fit.py` script that uses Hydra for configuration, supports multiple models (DeepFM, DIN, DLRM), and integrates with PyTorch Lightning and Weights & Biases for logging and early stopping. (`projects/recsys-ranking/src/fit.py` [projects/recsys-ranking/src/fit.pyR1-R138](diffhunk://#diff-720a9440ddc021279d27623520ae1bdc3505b0c39304ad744316ed24d8c421f3R1-R138))

**Model Implementation Improvements:**

* Refactored DeepFM and DIN models and modules to accept configuration parameters for hidden layer sizes, dropout, and normalization, improving flexibility and clarity. Also added detailed docstrings and usage examples to both models. (`projects/recsys-ranking/src/models/deepfm.py` [[1]](diffhunk://#diff-65d0444344eed84ff2ec4e7df2fcdf54b64dd7efa8f1b494515e9531b32be913L30-R106) [[2]](diffhunk://#diff-65d0444344eed84ff2ec4e7df2fcdf54b64dd7efa8f1b494515e9531b32be913R150-R153) [[3]](diffhunk://#diff-65d0444344eed84ff2ec4e7df2fcdf54b64dd7efa8f1b494515e9531b32be913R166-R169) [[4]](diffhunk://#diff-65d0444344eed84ff2ec4e7df2fcdf54b64dd7efa8f1b494515e9531b32be913L150-R183); `projects/recsys-ranking/src/models/din.py` [[5]](diffhunk://#diff-7215557d9304ffb099d3a8e74a4b0e62cc18086b07a7c567685eef782723a1e5L40-R49) [[6]](diffhunk://#diff-7215557d9304ffb099d3a8e74a4b0e62cc18086b07a7c567685eef782723a1e5L53-R62) [[7]](diffhunk://#diff-7215557d9304ffb099d3a8e74a4b0e62cc18086b07a7c567685eef782723a1e5L65-R82) [[8]](diffhunk://#diff-7215557d9304ffb099d3a8e74a4b0e62cc18086b07a7c567685eef782723a1e5L129-R148) [[9]](diffhunk://#diff-7215557d9304ffb099d3a8e74a4b0e62cc18086b07a7c567685eef782723a1e5L222-R261) [[10]](diffhunk://#diff-7215557d9304ffb099d3a8e74a4b0e62cc18086b07a7c567685eef782723a1e5L233-R275)

* Updated model module imports and exports for consistency and modularity. (`projects/recsys-ranking/src/models/__init__.py` [projects/recsys-ranking/src/models/__init__.pyR1-R5](diffhunk://#diff-2cc486b198dfadad46ee11f56a5cf8982aaeaffc03158d1120832abddf3f8023R1-R5))

**Constants and Typing:**

* Added a constant for evaluation negative sample size and updated `__all__` for clarity. (`projects/recsys-ranking/src/const/__init__.py` [projects/recsys-ranking/src/const/__init__.pyR1-R5](diffhunk://#diff-397275e9a4dd2c58ea39b7a0c2dc2ec7d1255601701bc57b4ece76556f7ad3e7R1-R5))

### CODE REVIEW

**Comment by haru-256** [comment]
@codex review

File: projects/recsys-ranking/src/fit.py [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+            item_pad_idx=SpecialItemIndex.PAD,
+            category_pad_idx=SpecialCategoryIndex.PAD,
+            max_seq_len=cfg.data.max_seq_len,
+            # optimizer
+            optimizer_params=optimizer_params,
+            # eval
+            eval_top_k=cfg.data.eval_top_k,
+        )
+    else:
+        raise NotImplementedError(f"{cfg.model.name=} is not supported")
```
[the remark this answers is not in the document]
Comment (haru-256) [reply projects/recsys-ranking/src/fit.py:105]: /gemini

please explain the following point deital

> perhaps using a factory pattern or a dictionary mapping model names to their respective classes and configurations. This would reduce code duplication and improve readability.
